fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1501,7 +1517,10 @@ public static IndexState writeOnlyIfTooManyRecordsForRebuild(long recordCount, b
                                                     @Nonnull StoreExistenceCheck existenceCheck,
                                                     @Nonnull CompletableFuture<Void> metaDataPreloadFuture) {
         CompletableFuture<Void> subspacePreloadFuture = preloadSubspaceAsync();
-        CompletableFuture<RecordMetaDataProto.DataStoreInfo> storeHeaderFuture = getStoreStateCache().get(this, existenceCheck).thenApply(storeInfo -> {
+
+        CompletableFuture<FDBRecordStoreStateCacheEntry> cacheEntryFuture = metaDataPreloadFuture.thenCompose(vignore ->
+                getStoreStateCache().get(this, existenceCheck));
```

Comment:
Ok, I remember. we need the record meta-data to convert subspace tuple key to index name. Also, later (when I send out its pr) we may need this when we store the index state at index subspace key rather than index name.